pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -105,17 +107,17 @@ func (r *Reconciler) reconcileBackupBucket(
 		return fmt.Errorf("could not update status after reconciliation start: %w", updateErr)
 	}
 
-	gardenSecret, err := kubernetesutils.GetSecretByObjectReference(gardenCtx, r.GardenClient, backupBucket.Spec.CredentialsRef) // TODO(vpnachev): Add support for WorkloadIdentity
+	backupCredentials, err := kubernetesutils.GetCredentialsByObjectReference(gardenCtx, r.GardenClient, *backupBucket.Spec.CredentialsRef)
 	if err != nil {
-		log.Error(err, "Failed to get backup secret", "credentialsRef", backupBucket.Spec.CredentialsRef)
-		r.Recorder.Eventf(backupBucket, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, "Failed to get backup secret from credentials reference %v: %w", backupBucket.Spec.CredentialsRef, err)
+		log.Error(err, "Failed to get backup credentials", "credentialsRef", backupBucket.Spec.CredentialsRef)
+		r.Recorder.Eventf(backupBucket, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, "Failed to get backup credentials from reference %v: %w", backupBucket.Spec.CredentialsRef, err)
 		return err
 	}
 
-	if !controllerutil.ContainsFinalizer(gardenSecret, gardencorev1beta1.ExternalGardenerName) {
-		log.Info("Adding finalizer to secret", "secret", client.ObjectKeyFromObject(gardenSecret))
-		if err := controllerutils.AddFinalizers(gardenCtx, r.GardenClient, gardenSecret, gardencorev1beta1.ExternalGardenerName); err != nil {
-			return fmt.Errorf("failed to add finalizer to backup secret: %w", err)
+	if !controllerutil.ContainsFinalizer(backupCredentials, gardencorev1beta1.ExternalGardenerName) {
+		log.Info("Adding finalizer to backup credentials", "kind", backupCredentials.GetObjectKind().GroupVersionKind().Kind, "key", client.ObjectKeyFromObject(backupCredentials))
```

Comment:
Does it make sense to be more explicit here?
```suggestion
		log.Info("Adding finalizer to backup credentials", "gvk", backupCredentials.GetObjectKind().GroupVersionKind(), "key", client.ObjectKeyFromObject(backupCredentials))
```

Reply:
Actually, I did it this way at some point of time but I did not like how GVK is [stringified](https://github.com/kubernetes/kubernetes/blob/393ecc29a5add2f936805af21290a1960c04056d/staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L161-L163), but that is my personal perception and I am open to change it.  

```go
{"gvk": "security.gardener.cloud/v1alpha1, Kind=WorkloadIdentity", ....}
```

Reply:
Yes, I also thought of that, but I still thought that it is worth the question if the "gvk" variant is better. Up to you, I am not against the other option as well since we only cover `Secret` and `WorkloadIdentity` and "kind" should be enough to identity them.

Reply:
Done with 8d92457565648459054806310f9527aac3f337e0